cpp/oneapi/dal/algo/pca/common.cpp

```diff
@@ -19,49 +19,62 @@
 
 namespace oneapi::dal::pca {
 
-class detail::descriptor_impl : public base {
+template <>
+class detail::descriptor_impl<task::dim_reduction> : public base {
 public:
     std::int64_t component_count = -1;
     bool is_deterministic = false;
 };
 
-class detail::model_impl : public base {
+template <>
+class detail::model_impl<task::dim_reduction> : public base {
 public:
     table eigenvectors;
 };
 
 using detail::descriptor_impl;
 using detail::model_impl;
 
-descriptor_base::descriptor_base() : impl_(new descriptor_impl{}) {}
+template <typename Task>
+descriptor_base<Task>::descriptor_base() : impl_(new descriptor_impl{}) {}
 
-std::int64_t descriptor_base::get_component_count() const {
+template <typename Task>
+std::int64_t descriptor_base<Task>::get_component_count() const {
```

Comment:
It depends on field presence. It would be better to replace "Task" with "task::dim_reduction"

Reply:
fixed